src/utility/svd.jl

```diff
@@ -438,13 +438,21 @@ function ChainRulesCore.rrule(
     return (U, S, V, info), tsvd!_itersvd_pullback
 end
 
-# scalar inverses with a cutoff tolerance and Lorentzian broadening
-function _safe_inv(x, tol, ε=0)
-    if abs(x) < tol
-        return zero(x)
-    else
-        return iszero(ε) ? inv(x) : _lorentz_broaden(x, ε)
+# scalar inverses with a cutoff tolerance
+_safe_inv(x, tol) = abs(x) < tol ? zero(x) : inv(x)
+
+# compute inverse singular value difference contribution to SVD gradient with broadening ε
+function _broadened_inv_S(S::AbstractVector{T}, tol, ε=0) where {T}
+    F = Matrix{T}(undef, length(S), length(S))
```

Comment:
```suggestion
    F = similar(S, (axes(S, 1), axes(S, 1)))
```
I'm okay with this being hardcoded to `length` and `Matrix` as well, but then the signature should probably be `Vector{T}` instead of `AbstractVector` (or at least `DenseVector`).

Reply:
I'll leave the `AbstractVector` since that is also the type restriction from `svd_pullback!` and then rather use the `similar` constructor